src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -169,7 +166,12 @@ public OAuthHandler(HttpClient backchannel)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var properties = new AuthenticationProperties(context.Properties);
+            var properties = new AuthenticationProperties(context.Properties)
+            {
+                // Review: should we add an option to configure the state lifetime?
+                ExpiresUtc = DateTimeOffset.UtcNow + TimeSpan.FromHours(1)
```

Comment:
Will do. I'll also use the `SystemClock` property added by @Tratcher in his recent PR to both the OAuth2 and the OIDC middleware.